src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,109 @@
+            var buildItems = GetContentForPattern(collection, managedCodeConventions.Patterns.MSBuildFiles);
+
+            var libFrameworks = GetGroupFrameworks(libItems).ToArray();
+            var refFrameworks = GetGroupFrameworks(refItems).ToArray();
+            var buildFrameworks = GetGroupFrameworks(buildItems).ToArray();
+
+            if (libFrameworks.Length == 0 && refFrameworks.Length == 0)
+            {
+                //if you can't find the ref and lib folder, then find the build folder
+                if (buildFrameworks.Length != 0)
+                {
+                    //if you can find any folders other than native or any, raise an NU5127
+                    if (buildFrameworks.Any(t => (FrameworkConstants.DotNetAll.Satisfies(t) || FrameworkNameValidatorUtility.IsValidFrameworkName(t))
+                        && t.GetShortFolderName() != "dotnet"
+                        && t.GetShortFolderName() != "native"))
+                    {
+                        var possibleFrameworks = buildFrameworks.
+                            Where(t => t.IsSpecificFramework && t.GetShortFolderName() != "dotnet" && t.GetShortFolderName() != "native").
+                            Select(t => t.GetShortFolderName()).ToArray();
+
+                        (var tfmNames, var suggestedDirectories) = GenerateWarningString(possibleFrameworks);
+                        
+                        var issue = new List<PackagingLogMessage>();
+                        issue.Add(PackagingLogMessage.CreateWarning(string.Format(MessageFormat, tfmNames, suggestedDirectories),
+                            NuGetLogCode.NU5127));
+                        return issue;
+                    }
+                }
+            }
+
+            return Array.Empty<PackagingLogMessage>();
+        }
+
+        private (string, string) GenerateWarningString(string[] possibleFrameworks)
+        {
+            string tfmNames = possibleFrameworks.Length > 1
+                ? string.Join(", ", possibleFrameworks, 0, possibleFrameworks.Count() - 1)
+                : possibleFrameworks[0];
+
+            string suggestedDirectories = possibleFrameworks.Length > 1
+                ? CreateDirectories(possibleFrameworks)
+                : string.Format("-lib/{0}/_._", possibleFrameworks[0]);
+            
+            return (tfmNames, suggestedDirectories);
+        }
+
+        private static IEnumerable<ContentItemGroup> GetContentForPattern(ContentItemCollection collection, PatternSet pattern)
+        {
+            return collection.FindItemGroups(pattern);
+        }
+
+        private static IEnumerable<NuGetFramework> GetGroupFrameworks(IEnumerable<ContentItemGroup> groups)
+        {
+            return groups.Select(e => ((NuGetFramework)e.Properties["tfm"]));
+        }
+
+        private static string CreateDirectories(string[] possibleFrameworks)
```

Comment:
How about `GenerateDirectoriesMessage`. The current name sounds like it should create directories on the filesystem.